### PR TITLE
Fix available equipement and add tests

### PR DESCRIPTION
**Pull request by Prolas**
* Fixed a bug that displayed all equipment and users as available when adding a flight
* Added tests

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt [hunk 1]
```diff
@@ -154,6 +155,108 @@ class FlightsViewModelTest {
...
+      assertEquals(expectedBaskets, currentBaskets.value.sortedBy { it.id })
+      assertEquals(expectedVehicles, currentVehicles.value.sortedBy { it.id })
+      assertEquals(expectedUsers, currentUsers.value.sortedBy { it.id })
+
+      viewModelAdmin.setDateAndTimeSlot(dbSetup.date1, TimeSlot.AM)
+
+      viewModelAdmin.refreshAvailableBalloons().join()
+      viewModelAdmin.refreshAvailableBaskets().join()
+      viewModelAdmin.refreshAvailableVehicles().join()
+      viewModelAdmin.refreshAvailableUsers().join()
```
Comment (Baribal02) [inline app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt:195]: The refresh are not necessary as it is already done in setDateAndTimeSlot but you need to wait that these coroutines are finished (which is what your code does in a way).

File: app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt [hunk 2]
```diff
@@ -154,6 +155,108 @@ class FlightsViewModelTest {
...
+      assertEquals(expectedBaskets, currentBaskets.value.sortedBy { it.id })
+      assertEquals(expectedVehicles, currentVehicles.value.sortedBy { it.id })
+      assertEquals(expectedUsers, currentUsers.value.sortedBy { it.id })
+
+      viewModelAdmin.setDateAndTimeSlot(dbSetup.date2, dbSetup.date2TimeSlotInverse)
+
+      viewModelAdmin.refreshAvailableBalloons().join()
+      viewModelAdmin.refreshAvailableBaskets().join()
+      viewModelAdmin.refreshAvailableVehicles().join()
+      viewModelAdmin.refreshAvailableUsers().join()
```
Comment (Baribal02) [inline app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt:243]: Same as before

**Review by Baribal02 — APPROVED** [review]
LGTM

File: app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt (hunk 1 above)
Comment (Prolas) [reply app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt:195]: Yes I know. Do you have a better solution ?

**Pull request merged**